src/fan.mli

```diff
@@ -15,37 +15,37 @@ furnished to do so, subject to the following conditions:
 The above copyright notice and this permission notice shall be included in
 all copies or substantial portions of the Software. *)
 
-type t
+type 'a t
 
-val equal : t -> t -> bool
-(** The equality function for fan-out. *)
+val equal : 'a t -> 'a t -> bool
+(** The equality function for fanouts. *)
 
-val v : hash_size:int -> entry_size:int -> int -> t
+val v : hash_size:int -> entry_size:int -> int -> [ `Write ] t
 (** [v ~hash_size ~entry_size n] creates a fan_out for an index with [hash_size]
     and [entry_size], containing [n] elements. *)
 
-val nb_fans : t -> int
+val nb_fans : 'a t -> int
 (** [nb_fans t] is the number of fans in [t]. *)
 
-val search : t -> int -> int64 * int64
+val search : [ `Read ] t -> int -> int64 * int64
 (** [search t hash] is the interval of offsets containing [hash], if present. *)
 
-val update : t -> int -> int64 -> unit
+val update : [ `Write ] t -> int -> int64 -> unit
 (** [update t hash off] updates [t] so that [hash] is registered to be at offset
     [off]. *)
 
-val finalize : t -> unit
-(** Finalizes the update of the fanout. This is mendatory before any [search]
+val finalize : [ `Write ] t -> [ `Read ] t
+(** Finalizes the update of the fanout. This is mandatory before any [search]
     query. *)
```

Comment:
```suggestion
(** Finalizes the update of the fanout. *)
```

Not really needed now that the types imply it, but keep it if you want.

Reply:
Indeed, but I thought it wasn't that costly to keep it there. Happy to remove it if that feels odd though.

Reply:
Np; let's keep it.